Server/models/Check.js

```diff
@@ -31,4 +31,27 @@ const CheckSchema = mongoose.Schema(
   }
 );
 
+/**
+ * When a check is created, we should update the stauts of the associated monitor
+ * if it has changed.  Monitor starts with default status: true
+ */
+
+CheckSchema.pre("save", async function (next) {
+  const monitor = await mongoose.model("Monitor").findById(this.monitorId);
+  if (monitor.status !== this.status) {
```

Comment:
please put this whole code block inside an if statement for error handling purposes, like this:

```
if (monitor) {
// other if(s)
} else {
// report that no monitors are found
}
```

Reply:
Good call, I've added some error handling